libraries/tlv-account-resolution/src/account.rs

```diff
@@ -0,0 +1,320 @@
+//! Structs/enums for managing "required account state", ie. defining accounts
+//! required for your interface program, which can be  `AccountMeta`s - which
+//! have fixed addresses - or PDAs - which have addresses derived from a
+//! collection of seeds
+
+use {
+    crate::{error::AccountResolutionError, seeds::Seed},
+    bytemuck::{Pod, Zeroable},
+    solana_program::{
+        account_info::AccountInfo, instruction::AccountMeta, program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    spl_type_length_value::pod::PodBool,
+};
+
+/// `Pod` type for defining a required account in a validation account.
+///
+/// This can either be a standard `AccountMeta` or a `AccountMetaPda`.
+/// Can be used in TLV-encoded data.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+pub struct PodAccountMeta {
+    /// Discriminator to tell whether this represents a standard
+    /// `AccountMeta` or an `AccountMetaPda`
+    pub discriminator: u8,
+    /// This `address_config` field can either be the pubkey of the account
+    /// or the seeds used to derive the pubkey from provided inputs
+    pub address_config: [u8; 32],
```

Comment:
Clever! Will produce an implementation. This should make things much cleaner

Reply:
Actually I tried this at first. Maybe I missed a possible avenue this could go.

When you try to put that enum into `PodAccountMeta`, you end up with errors implementing `Pod` and `Zeroable` on the enum:

```rust
#[repr(C)]
#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
pub enum AddressConfig {
    Pubkey(Pubkey),
    Seeds([u8; 32]),
}

#[repr(C)]
#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
pub struct PodAccountMeta2 {
    pub address_config: AddressConfig,
}
```

```shell
Deriving Pod is not supported for enums
```

We're also on the latest version. Any ideas?

Reply:
It's actually expected behavior -- enums are not `Pod`, because `Pod` means that *any* bytes are valid, but if you have an enum with two variants, then `0` and `1` are valid for that byte, but what if you get `120`? So you might need to derive your own in-place deserializer, or implement `Pod` by hand. Or do the `unsafe` thing

Reply:
ahh ok, sounds fun. I'll give it a go. Likely going to avoid `unsafe` if I can

Reply:
Actually, with some of the other changes, I was able to formulate a simpler solution to combining these. Let me know what you think. It doesn't involve an enum, just two `new_with_*` functions